File: src/main/java/ink/rubi/coffee/controller/DataSourceController.java
package ink.rubi.coffee.controller;

import ink.rubi.coffee.po.holder.DataSourceConfigHolder;
import javafx.fxml.FXML;
import javafx.scene.control.TextField;
import javafx.scene.layout.GridPane;
import lombok.Getter;
import lombok.extern.slf4j.Slf4j;

import java.net.URL;
import java.util.ResourceBundle;

/**
 * @author : Rubi
 * @version : 2019-05-17 23:27 下午
 */
@Getter
@Slf4j
public class DataSourceController implements IController<DataSourceConfigHolder> {

    @FXML
    private GridPane dataSourcePage;
    @FXML
    private TextField url, driverName, username, password, schemaName;

    private MainController mainController;

    @Override
    public void initialize(URL location, ResourceBundle resources) {

    }

    @Override
    public DataSourceConfigHolder getConfigHolder() {
        return new DataSourceConfigHolder()
                .setUrl(url.getText())
                .setSchemaName(schemaName.getText())
                .setDriverName(driverName.getText())
                .setUsername(username.getText())
                .setPassword(password.getText());

    }

    @Override
    public void init(MainController mainController) {
        this.mainController = mainController;
    }

    @Override
    public void flushConfig(DataSourceConfigHolder dataSourceConfig) {
        url.setText(dataSourceConfig.getUrl());
        driverName.setText(dataSourceConfig.getDriverName());
        username.setText(dataSourceConfig.getUsername());
        password.setText(dataSourceConfig.getPassword());
        schemaName.setText(dataSourceConfig.getSchemaName());
    }


}

File: src/main/java/ink/rubi/coffee/controller/DataSourceController.kt
package ink.rubi.coffee.controller

import ink.rubi.coffee.po.holder.DataSourceConfigHolder
import javafx.fxml.FXML
import javafx.fxml.Initializable
import javafx.scene.control.TextField
import javafx.scene.layout.GridPane
import java.net.URL
import java.util.*

/**
 * @author : Rubi
 * @version : 2019-05-17 23:27 下午
 */
class DataSourceController : IController<DataSourceConfigHolder>, Initializable {

    @FXML
    private lateinit var dataSourcePage: GridPane
    @FXML
    private lateinit var url: TextField
    @FXML
    private lateinit var driverName: TextField
    @FXML
    private lateinit var username: TextField
    @FXML
    private lateinit var password: TextField
    @FXML
    private lateinit var schemaName: TextField

    private var mainController: MainController? = null

    override fun initialize(location: URL, resources: ResourceBundle?) {
        defaultShow()
    }

    override fun getConfigHolder(): DataSourceConfigHolder {
        return DataSourceConfigHolder(schemaName.text, url.text, driverName.text, username.text, password.text)
    }

    override fun inject(mainController: MainController) {
        this.mainController = mainController
    }

    override fun defaultShow() {
        url.text = "jdbc:mysql://localhost:3306/table_name?useUnicode=true&useSSL=false&characterEncoding=utf8"
        driverName.text = "com.mysql.jdbc.Driver"
        username.text = "root"
        password.text = ""
        schemaName.text = ""
    }


    override fun flushConfig(holder: DataSourceConfigHolder) {
        url.text = holder.url
        driverName.text = holder.driverName
        username.text = holder.username
        password.text = holder.password
        schemaName.text = holder.schemaName
    }


}
